axum-macros/src/lib.rs

```diff
@@ -540,6 +540,29 @@ pub fn derive_from_request_parts(item: TokenStream) -> TokenStream {
 /// }
 /// ```
 ///
+/// # Limitations
+///
+/// This macro does not work for associated functions — functions defined in an impl block.
+///
+/// ```
+/// use axum::{debug_handler, extract::Path};
+/// struct App {}
+/// #[debug_handler]
+/// impl App {
+///     pub async fn handler(Path(_): Path(String)) {}
+/// }
+/// ```
```

Comment:
```suggestion
/// ```compile_fail
/// use axum::{debug_handler, extract::Path};
///
/// struct App {}
///
/// impl App {
///     #[debug_handler]
///     pub async fn handler(Path(_): Path(String)) {}
/// }
/// ```
```